### ISSUE TITLE
Lint does not detect nested <tr>, <td> and <th> tags

### ISSUE THREAD

**Issue by petdance** [petdance/html-lint#33]
```
$ cat foo.html
<html>
    <head><title>foo</title></head>
    <body>
        <table>
            <tr>
                <tr> <!-- This should be an error -->
                    <td>
                        <td></td> <!-- This should be an error -->
                    </td>
                    <th>
                        <th></th> <!-- This should be an error -->
                    </th>
                </tr>
            </tr>
        </table>
    </body>
</html>
$ weblint foo.html
$
```

The `<td>`, `<tr>` and `<th>` should not be able to nest like that.
